app/uk/gov/hmrc/tradergoodsprofilesrouter/models/request/eis/withdrawAdvice/withdrawAdvice.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.tradergoodsprofilesrouter.models.request.eis.withdrawAdvice
+
+import play.api.libs.json.{JsPath, Json, OFormat, Reads, Writes}
+import uk.gov.hmrc.tradergoodsprofilesrouter.service.DateTimeService.DateTimeFormat
+
+import java.time.Instant
+
+object withdrawAdvice {
+
+  case class RequestCommon(clientID: Option[String])
+
+  object RequestCommon {
+    implicit val format: OFormat[RequestCommon] = Json.format[RequestCommon]
+  }
+
+  case class WithdrawDetail(withdrawDate: Instant, withdrawReason: Option[String])
+
+  object WithdrawDetail {
+    import play.api.libs.functional.syntax._
+
+    implicit val format: Reads[WithdrawDetail] = Json.reads[WithdrawDetail]
```

Comment:
does this work if caller wants to remove the withdrawReason? like if caller passes `{withdrawReason: null}` to remove the same?

Reply:
This should never happens. The withdraw reason should never be null from the trader-goods-profiles. This should be either an empty json (if not present) or a valid json if present. If I don't mistaken we have the problem with null in json only in the EIS response because they wrongly send null as value json